registry/remote.go

```diff
@@ -0,0 +1,64 @@
+package registry
+
+import (
+	"context"
+	"encoding/json"
+	manifest "github.com/docker/distribution/manifest/schema1"
+	dockerregistry "github.com/heroku/docker-registry-client/registry"
+	"time"
+)
+
+type Remote interface {
+	Tags(img Image) ([]string, error)
+	Manifest(img Image) (Image, error)
+	Cancel()
+}
+
+type remote struct {
+	client dockerRegistryInterface
+	cancel context.CancelFunc
+}
+
+func newRemote(client *dockerregistry.Registry, cancel context.CancelFunc) Remote {
+	return &remote{
+		client: client,
+		cancel: cancel,
+	}
+}
+
+func (rc *remote) Tags(id Image) (_ []string, err error) {
+	return rc.client.Tags(id.OrgRepo())
+}
+
+func (rc *remote) Manifest(img Image) (Image, error) {
```

Comment:
Its the remote's responsibility to convert the raw Manifest into a domain object (an image).

If you want more distinction, then we could do the same as above? Change it to:

`+func (rc *remote) Manifest(repo Repository, ref string) (Image, error)`